lib/spack/docs/module_file_tutorial.rst

```diff
@@ -0,0 +1,960 @@
+     netlib-scalapack/2.0.2-netlib    netlib-scalapack/2.0.2-openblas (D)
+
+  -------------------------------------------------------------------- /home/mculpo/wdir/spack/share/spack/lmod/linux-Ubuntu14-x86_64/gcc/6.2.0 ---------------------------------------------------------------------
+     binutils/2.27    curl/7.50.3    hwloc/1.11.4           libtool/2.4.6    lzo/2.09           netlib-lapack/3.6.1    openssl/1.0.2j              py-scipy/0.18.1-openblas    util-macros/1.19.0
+     bison/3.0.4      expat/2.2.0    libarchive/3.2.1       libxml2/2.9.4    m4/1.4.17          nettle/3.2             pkg-config/0.29.1           py-setuptools/25.2.0        xz/5.2.2
+     bzip2/1.0.6      flex/2.6.0     libpciaccess/0.13.4    lz4/131          mpich/3.2   (L)    openblas/0.2.19        py-nose/1.3.7               python/2.7.12               zlib/1.2.8
+     cmake/3.6.1      gmp/6.1.1      libsigsegv/2.10        lzma/4.32.7      ncurses/6.0        openmpi/2.0.1          py-numpy/1.11.1-openblas    sqlite/3.8.5
+
+  ----------------------------------------------------------------------- /home/mculpo/wdir/spack/share/spack/lmod/linux-Ubuntu14-x86_64/Core -----------------------------------------------------------------------
+     gcc/6.2.0 (L)
+
+  $ module load openblas netlib-scalapack/2.0.2-openblas
+  $ module list
+
+  Currently Loaded Modules:
+    1) gcc/6.2.0   2) mpich/3.2   3) openblas/0.2.19   4) netlib-scalapack/2.0.2-openblas
+
+  $ module load openmpi
+
+  Lmod is automatically replacing "mpich/3.2" with "openmpi/2.0.1"
+
+
+  Due to MODULEPATH changes the following have been reloaded:
+    1) netlib-scalapack/2.0.2-openblas
+
+This layout is already a great improvement over the usual non-hierarchical layout,
+but it still has an asymmetry: ``LAPACK`` providers are semantically the same as ``MPI``
+providers, but they are still not part of the hierarchy. One possible solution is shown next.
+
+.. Activate lmod and turn the previous modifications into lmod:
+   Add core compilers
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Extend the hierarchy to other virtual providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning::
+  This is an experimental feature
+    Having a hierarchy deeper than ``Core``/``Compiler``/``MPI`` is an experimental
+    feature, still not fully supported by ``module spider``,
+    see `here <https://github.com/TACC/Lmod/issues/114>`_. Furthermore its use
+    with hierarchies more complex than ``Core``/``Compiler``/``MPI``/``LAPACK``
+    has not been thoroughly tested in production environments.
+
+Spack permits to generate Lua hierarchical module files where users
+can add an arbitrary list of virtual providers to the triplet
+``Core``/``Compiler``/``MPI``. A configuration file like:
+
+.. code-block:: yaml
+
+  modules:
+    enable::
+      - lmod
+    lmod:
+      core_compilers:
+        - 'gcc@4.8'
+      hierarchical_scheme:
+        -	lapack
```

Comment:
No... but `py-charm` transforms spaces into tabs for rst files. I missed this one.